classes/local/datasource/roles.php

```diff
@@ -312,10 +324,15 @@ function($v) {
      * @return string|null
      */
     public function get_instance_cache_key(): ?string {
+        $cacherev = 0;
         if (!empty($this->cms->get('id'))) {
             $this->cms->read();
+            $cacherev = $this->cms->get_custom_data('roles_course_role_cache_rev');
+            if (!$cacherev) {
```

Comment:
Again here we should be explicit what type of falsehood we are testing, is it `$cacherev === false`? or `$cacherev === 0`? or `is_null($cacherev)`?